### PR TITLE
Add initial v1.2 draft spec

### PR DESCRIPTION
**Pull request by steventux**
## Context

We are adding the ability to reject applications for one or more pre-defined reasons. This will be released in v1.2 of the Vendor API.

## Changes proposed in this pull request

Adds a draft spec for v1.2 with proposed endpoints.

image

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: config/vendor_api/draft.yml [hunk 1]
```diff
@@ -154,591 +55,32 @@ paths:
...
-          description: API version that produced this response
-          example: 'v1.1'
-        timestamp:
-          type: string
-          format: date-time
-          description: When this response was generated
-          example: 2021-12-01T11:03:12Z
-        total_count:
+        code:
           type: integer
```
Comment (carlosmartinez) [inline config/vendor_api/draft.yml:79]: might this be a string?

File: config/vendor_api/draft.yml [hunk 2]
```diff
@@ -154,591 +55,32 @@ paths:
...
-            application/json:
-              schema:
-                "$ref": "#/components/schemas/SingleApplicationResponse"
-        '401':
-          "$ref": "#/components/responses/Unauthorized"
-        '404':
-          "$ref": "#/components/responses/NotFound"
-        '422':
-          "$ref": "#/components/responses/UnprocessableEntity"
+                "$ref": "#/components/schemas/ListResponse"
```
Comment (carlosmartinez) [inline config/vendor_api/draft.yml:68]: presumably the response will be a hash of codes and labels, such as 
```
[{\"code\":\"A1\",\"label\":\"Nah mate\"},{\"code\":\"A2\",\"label\":\"Not a good vibe\"}]
```
is this covered by `ListResponse`?

**Comment by carlosmartinez** [comment]
looks good, just a couple of small questions

File: config/vendor_api/draft.yml (hunk 1 above)
Comment (steventux) [reply config/vendor_api/draft.yml:79]: Yeah string is definitely more flexible

File: config/vendor_api/draft.yml (hunk 2 above)
Comment (steventux) [reply config/vendor_api/draft.yml:68]: Possibly the simpler and definitely lazier approach was to return something like

```
[
  "A1: Nah mate",
  "A2: Not a good vibe"
]
```

File: config/vendor_api/draft.yml (hunk 1 above)
Comment (steventux) [reply config/vendor_api/draft.yml]: [Now a string](https://github.com/DFE-Digital/apply-for-teacher-training/pull/7254/commits/9560905156ab9c380482eba1f0b16df04e8e2cb1#diff-409b89434fef4cef9730cceeef56883e0f6fd7ac8ec20fbaa2d10a251931d47dR79)

**Review by carlosmartinez — APPROVED** [review]

**Review by JR-G — APPROVED** [review]

**Pull request merged**